File: benchmarks/tasks/data_versioned_register/starter.py
from dataclasses import dataclass

@dataclass(frozen=True)
class Entry:
    key: int
    version: int
    value: int

@dataclass(frozen=True)
class Report:
    accepted: int
    stale: int
    checksum: int

def put(table: dict[int, int], key: int, value: int) -> dict[int, int]:
    table[key] = value
    return table

def solve(rows: list[Entry]) -> Report:
    versions: dict[int, int] = {}
    values: dict[int, int] = {}
    accepted: int = 0
    stale: int = 0
    checksum: int = 0
    for row in rows:
        if (row.version >= versions.get(row.key, (-1))):
            checksum = (checksum + ((row.key + 1) * (row.value - values.get(row.key, 0))))
            updated_value_0: int = row.value
            values = put(values, row.key, updated_value_0)
            updated_value_1: int = row.version
            versions = put(versions, row.key, updated_value_1)
            accepted = (accepted + 1)
        else:
            stale = (stale + 1)
    return Report(accepted, stale, checksum)

File: benchmarks/tasks/data_reservation_gate/starter.go
package benchmark

type Entry struct {
	Party int64
	Seats int64
}

type Report struct {
	Accepted int64
	Rejected int64
	Occupied int64
}

func put(table map[int64]int64, key int64, value int64) map[int64]int64 {
	table[key] = value
	return table
}

func read(table map[int64]int64, key int64, fallback int64) int64 {
	value, exists := table[key]
	if exists {
		return value
	}
	return fallback
}

func solve(rows []Entry, limit int64) Report {
	var parties map[int64]int64 = make(map[int64]int64)
	var accepted int64 = 0
	var rejected int64 = 0
	var occupied int64 = 0
	for _, row := range rows {
		var invalid bool = ((row.Seats <= 0) || (read(parties, row.Party, 0) == 1))
		if invalid || ((occupied + row.Seats) >= limit) {
			rejected = (rejected + 1)
		} else {
			accepted = (accepted + 1)
			occupied = (occupied + row.Seats)
			var updated_value_0 int64 = 1
			parties = put(parties, row.Party, updated_value_0)
		}
	}
	return Report{Accepted: accepted, Rejected: rejected, Occupied: occupied}
}
